packages/type-utils/typings/typescript.d.ts

```diff
@@ -22,6 +22,10 @@ declare module 'typescript' {
      * Return the type of the given property in the given type, or undefined if no such property exists
      */
     getTypeOfPropertyOfType(type: Type, propertyName: string): Type | undefined;
+    /**
+     * Return the awaited type of the given type.
```

Comment:
Maybe add a TODO that this should no longer be necessary in a future TS version and link the TS issue?

Reply:
Sure!  https://github.com/typescript-eslint/typescript-eslint/pull/8693/commits/fcca5ecc161696a97ee207db4839056e46c438eb